src/components/BeautifulProgressIndicator.vue

```diff
@@ -1,22 +1,35 @@
 <template>
   <div class="container">
-    <p class="ratio-label">{{earned}} / {{required}}</p>
-    <p class="planned-label" v-if="planned > 0">{{planned}} noch geplant</p>
+    <p class="ratio-label">
+      {{ earned }} / {{ required }}
+    </p>
+    <p
+      v-if="planned > 0"
+      class="planned-label"
+    >
+      {{ planned }} noch geplant
+    </p>
 
     <div class="progress-container">
-      <div class="planned-progress" v-bind:style="{ width: plannedProgress }" />
+      <div
+        class="planned-progress"
+        :style="{ width: plannedProgress }"
+      />
       <div
         class="earned-progress"
-        v-bind:style="{
+        :style="{
           width: earnedProgress,
           'background-color': color,
-        }" />
+        }"
+      />
     </div>
   </div>
 </template>
 
-<script>
-export default {
+<script lang="ts">
+import { defineComponent } from 'vue'
+
+export default defineComponent({
```

Comment:
this is fine, but if you're interested, take a look at vue setup scripts, they're amazing!

Reply:
Sounds nice, but I'll leave it like this for now, since focus of this PR was on TS ^^;